cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionConsumerResult.java

```diff
@@ -35,7 +36,7 @@ public PartitionConsumerState getPartitionConsumerState() {
     return partitionConsumerState;
   }
 
-  public Iterator<Partition> getPartitionIterator() {
+  public List<PartitionDetail> getPartitions() {
```

Comment:
this is an API change. We need to keep the existing method getPartitionIterator().